lib/Installation/Partitioner/ExpertPartitionerPage.pm

```diff
@@ -141,4 +149,39 @@ sub press_ok_clone {
     send_key($self->{ok_clone_shortcut});
 }
 
+sub modify_create_new_partition_table {
+    my ($self, $item) = @_;
+    assert_screen(EXPERT_PARTITIONER_PAGE);
+    _select_system_view_section();
+    send_key_until_needlematch(SELECTED_HARD_DISKS, 'down');
+    send_key($self->{modify_hard_disk_shortcut});
+    send_key_until_needlematch(SELECTED_CREATE_NEW_TABLE, 'down');
+    send_key "ret";
+}
+
+sub select_msdos_partition_table {
+    my ($self) = @_;
+    assert_screen(NEW_PARTITION_TABLE_TYPE);
+    send_key($self->{select_msdos_shortcut});
+    send_key "ret";
+}
+
+sub confirm_deleting_current_devices {
+    my ($self) = @_;
+    assert_screen(DELETING_CURRENT_DEVICES);
+    send_key($self->{press_yes});
+}
+
+sub switch_to_partitions_tab {
+    my ($self) = @_;
+    assert_screen(EXPERT_PARTITIONER_PAGE);
+    send_key($self->{partitions_tab_shortcut});
+}
+
+sub select_primary_partition {
```

Comment:
same here, you can leave the function parameterized for next person to touch this code who would need to create an Extended Partition. for example, s/select_primary_partition/add_new_msdos_partition/.